src/test/enhetstester/kotlin/no/nav/familie/ba/sak/kjerne/arbeidsfordeling/ArbeidsfordelingServiceTest.kt

```diff
@@ -0,0 +1,113 @@
+        ArbeidsfordelingService(
+            arbeidsfordelingPåBehandlingRepository = arbeidsfordelingPåBehandlingRepository,
+            personopplysningGrunnlagRepository = personopplysningGrunnlagRepository,
+            personidentService = personidentService,
+            oppgaveService = oppgaveService,
+            loggService = loggService,
+            integrasjonClient = integrasjonClient,
+            personopplysningerService = personopplysningerService,
+            saksstatistikkEventPublisher = saksstatistikkEventPublisher,
+            tilpassArbeidsfordelingService = tilpassArbeidsfordelingService,
+            unleashService = unleashService,
+        )
+
+    @Nested
+    inner class FastsettBehandlendeEnhet {
+        @Test
+        fun `skal overstyre behandlende enhet fra NORG dersom enhet fra finnArbeidsfordelingForOppgave er en annen`() {
+            // Arrange
+            val behandling = lagBehandling()
+            val søker = lagPersonEnkel(PersonType.SØKER, behandling.fagsak.aktør)
+            val barn = lagPersonEnkel(PersonType.BARN)
+            val arbeidsfordelingsenhet =
+                Arbeidsfordelingsenhet(
+                    enhetId = BarnetrygdEnhet.MIDLERTIDIG_ENHET.enhetsnummer,
+                    enhetNavn = BarnetrygdEnhet.MIDLERTIDIG_ENHET.enhetsnavn,
+                )
+
+            val oppdatertArbeidsfordelingPåBehandling = mockk<ArbeidsfordelingPåBehandling>()
+
+            every {
+                arbeidsfordelingPåBehandlingRepository.finnArbeidsfordelingPåBehandling(behandling.id)
+            } returns null
+
+            every {
+                unleashService.isEnabled(FeatureToggleConfig.OPPRETT_SAK_PÅ_RIKTIG_ENHET_OG_SAKSBEHANDLER, false)
+            } returns true
+
+            every { personopplysningerService.hentPersoninfoEnkel(any()).adressebeskyttelseGradering } returns null
+
+            every {
+                personopplysningGrunnlagRepository
+                    .finnSøkerOgBarnAktørerTilAktiv(behandling.id)
+            } returns listOf(søker, barn)
+
+            every { integrasjonClient.hentBehandlendeEnhet(søker.aktør.aktivFødselsnummer()) } returns
+                listOf(
+                    arbeidsfordelingsenhet,
+                )
+
+            every {
+                tilpassArbeidsfordelingService.tilpassArbeidsfordelingsenhetTilSaksbehandler(
+                    arbeidsfordelingsenhet = arbeidsfordelingsenhet,
+                    navIdent = NavIdent(SikkerhetContext.hentSaksbehandler()),
+                )
+            } returns Arbeidsfordelingsenhet(enhetId = BarnetrygdEnhet.OSLO.enhetsnummer, enhetNavn = BarnetrygdEnhet.OSLO.enhetsnavn)
+
+            val arbeidsfordelingPåBehandlingSlot = slot<ArbeidsfordelingPåBehandling>()
+
+            every {
+                arbeidsfordelingPåBehandlingRepository.save(capture(arbeidsfordelingPåBehandlingSlot))
+            } returns oppdatertArbeidsfordelingPåBehandling
```

Comment:
Kan denne erstattes med `returnsArgument 0`? Da slipper man å mocke `oppdatertArbeidsfordelingPåBehandling`

```suggestion
            } returnsArgument 0
```